### PR TITLE
sync upload feature

### PR DESCRIPTION
**Pull request by chauchausoup**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -939,10 +940,39 @@ module.exports = {
         formData.append('remote_path', path);
         formData.append('client_json', JSON.stringify(client_json))
         formData.append('encrypt', encrypt)
+
         const response = await utils.postReq(url, formData, option);
         return response
     },
 
+    syncUploadObject: async function(feed,remote_path,allocation,client_json,encrypt,delay = 5, option=null){
```
Comment (shaktals) [inline index.js:948]: We have not enforced prettier on js-client-sdk, but even if we don't do it, it'd be nice to attend to some basic patterns:

```suggestion
    syncUploadObject: async function (feed, remote_path, allocation, client_json, encrypt,delay = 5, option = null) {
```
If you can do the same for the rest of the changes, it'd be great!

File: index.js [hunk 2]
```diff
@@ -1108,6 +1131,13 @@ module.exports = {
         return sig.serializeToHexStr();
     },
 
+    getShareInfo: async function (activeWallet) {
+        const url = zeroBoxUrl + Endpoints.ZEROBOX_SERVER_SHARE_INFO_ENDPOINT;
+        const clientSignature = this.getSign(activeWallet.id, activeWallet.secretKey);
+        const response = await utils.getShareInfo(url, clientSignature, activeWallet.id, activeWallet.client_key);
+        return response
+    },
```
Comment (shaktals) [inline index.js:1139]: Any reason to change the position of `getShareInfo`?

**Review by shaktals — APPROVED** [review]
Looks good! :+1:

File: index.js (hunk 2 above)
Comment (chauchausoup) [reply index.js:1139]: @shaktals I will come up to the PR once the /sync upload issue is worked on by the proxy server team.

**Pull request closed without merging**